### DISCUSSION TITLE
Modality combinations

### DISCUSSION THREAD

**Discussion by paul-reiners** [MIC-DKFZ/nnUNet#828] [Q&A] [answered]
Suppose I train a model where each subject has exactly one T1w image, exactly one T2w image, and a segmentation file.

Can I feed more than one T1 or T2 in per subject at a time for inference? If a subject has more than one T1 but only one T2, can it be run with the multiple T1s?

**Comment by FabianIsensee** [comment, marked as answer]
Modalities have to be the same for all training cases and have to be the same at test time as well.
nnU-Net learns a mapping from some input image(s) to a target segmentation, sothe problem formulation is

T1w + T2 -> segmentation

This cannot be changed at test time.
If you really want to experiment with this you can consider implementing a data augmentation that at training time randomly blackens out (sets to 0) one of the two inputs. Then the model should be able to somewhat handle the case where empty T1 or T2 images are given at test time
